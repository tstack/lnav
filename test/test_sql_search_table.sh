#! /bin/bash

export TZ=UTC
export YES_COLOR=1

run_cap_test ${lnav_test} -n \
    -c ';SELECT * FROM procstate_procs' \
    ${test_dir}/logfile_procstate.0

run_cap_test ${lnav_test} -n \
    -c ';SELECT *,log_body FROM vpx_lro_begin' \
    ${test_dir}/logfile_vpxd.0

run_cap_test ${lnav_test} -n \
    -c ";select * from vpx_lro_begin where log_line > 3 and lro_id = 'lro-846064'" \
    -c ";select * from vpx_lro_begin where lro_id = 'lro-846064'" \
    ${test_dir}/logfile_vpxd.0

run_cap_test ${lnav_test} -n \
    -c ";select * from procstate_procs where cmd_name = '[kthreadd]'" \
    -c ";select * from procstate_procs where cmd_name = '[kthreadd]'" \
    ${test_dir}/logfile_procstate.0

touch -t 202211030923 ${test_dir}/logfile_syslog.3

run_cap_test ${lnav_test} -n \
    -c ':create-search-table asl_mod ASL Module "(?<name>[^"]+)"' \
    -c ';SELECT * FROM asl_mod' \
    ${test_dir}/logfile_syslog.3
