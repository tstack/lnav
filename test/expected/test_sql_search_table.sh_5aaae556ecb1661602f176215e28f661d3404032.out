[1m[4mlog_line [0m[1m[4mlog_part [0m[1m[4m       log_time         [0m[1m[4m[7mlog_idle_msecs [0m[1m[4mlog_level [0m[1m[4mlog_mark [0m[1m[4mlog_comment [0m[1m[4mlog_tags [0m[1m[4mlog_annotations [0m[1m[4mlog_filters [0m[1m[4mmatch_index [0m[1m[4muser [0m[1m[4mpid [0m[1m[4m[7mcpu_pct [0m[1m[4m[7mmem_pct [0m[1m[4m[7mvsz [0m[1m[4m[7mrss [0m[1m[4mtty [0m[1m[4mstat [0m[1m[4mstart_time [0m[1m[4mcpu_time [0m[1m[4m   cmd     [0m[1m[4m cmd_name  [0m[1m[4mcmd_args [0m
       0   <NULL> 2022-06-02 00:01:01.000 [1m             0[0m info             0      <NULL>   <NULL>          <NULL>      <NULL>           1 root   2 [1m      0[0m [1m      0[0m [1m  0[0m [1m  0[0m ?   S    Jun01      0:00     [kthreadd] [kthreadd]   <NULL> 
[7m      12[0m[7m [0m[7m  <NULL>[0m[7m [0m[7m2022-06-02 00:02:01.000[0m[7m [0m[1m[7m         60000[0m[7m [0m[7minfo     [0m[7m [0m[7m       0[0m[7m [0m[7m     <NULL>[0m[7m [0m[7m  <NULL>[0m[7m [0m[7m         <NULL>[0m[7m [0m[7m     <NULL>[0m[7m [0m[7m          1[0m[7m [0m[7mroot[0m[7m [0m[7m  2[0m[7m [0m[1m[7m      0[0m[7m [0m[1m[7m      0[0m[7m [0m[1m[7m  0[0m[7m [0m[1m[7m  0[0m[7m [0m[7m?  [0m[7m [0m[7mS   [0m[7m [0m[7mJun01     [0m[7m [0m[7m0:00    [0m[7m [0m[7m[kthreadd][0m[7m [0m[7m[kthreadd][0m[7m [0m[7m  <N[0mULL> 
[1m[7m      30[0m[1m[7m [0m[1m[7m  <NULL>[0m[1m[7m [0m[1m[7m2022-06-02 00:03:01.000[0m[1m[7m [0m[1m[7m         60000[0m[1m[7m [0m[1m[7minfo     [0m[1m[7m [0m[1m[7m       0[0m[1m[7m [0m[1m[7m     <NULL>[0m[1m[7m [0m[1m[7m  <NULL>[0m[1m[7m [0m[1m[7m         <NULL>[0m[1m[7m [0m[1m[7m     <NULL>[0m[1m[7m [0m[1m[7m          1[0m[1m[7m [0m[1m[7mroot[0m[1m[7m [0m[1m[7m  2[0m[1m[7m [0m[1m[7m      0[0m[1m[7m [0m[1m[7m      0[0m[1m[7m [0m[1m[7m  0[0m[1m[7m [0m[1m[7m  0[0m[1m[7m [0m[1m[7m?  [0m[1m[7m [0m[1m[7mS   [0m[1m[7m [0m[1m[7mJun01     [0m[1m[7m [0m[1m[7m0:00    [0m[1m[7m [0m[1m[7m[kthreadd][0m[1m[7m [0m[1m[7m[kthreadd][0m[1m[7m [0m[1m[7m  <N[0m[1mULL> [0m
